io_scene_niftools/modules/nif_import/geometry/mesh/__init__.py

```diff
@@ -93,42 +93,42 @@ def import_mesh(self, n_block, b_obj):
             if vertex_attributes.u_vs:
                 uvs = [[vertex.uv for vertex in vertex_data]]
             if vertex_attributes.vertex_colors:
-                vertex_colors = [vertex.vertex_colors for vertex in vertex_data]
+                vertex_colors = [NifClasses.Color4.from_value(tuple(c / 255.0 for c in vertex.vertex_colors)) for vertex in vertex_data]
             if vertex_attributes.normals:
                 normals = [vertex.normal for vertex in vertex_data]
         elif isinstance(n_block, NifClasses.NiMesh):
-            # if it has a displaylist then we don't know how to process this NiMesh
+            # if it has a displaylist then the vertex data is encoded differently
             displaylist_data = n_block.geomdata_by_name("DISPLAYLIST", False, False)
             if len(displaylist_data) > 0:
                 displaylist = DisplayList(displaylist_data)
-                vertices_info, triangles = displaylist.create_mesh_data(n_block)
+                vertices_info, triangles, weights = displaylist.create_mesh_data(n_block)
```

Comment:
create is not a fitting name for this imo
get, unpack, extract or decode maybe?